### PR TITLE
Feature/orders

### PR DESCRIPTION
**Pull request by Shamskol**

### CODE REVIEW

File: api/order/orderRouter.js [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+ *                  name: 'Frank Martinez'
+ *                  avatarUrl: 'https://s3.amazonaws.com/uifaces/faces/twitter/hermanobrother/128.jpg'
+ *                - id: '013e4ab94d96542e791f'
+ *                  email: 'cathy@example.com'
+ *                  name: 'Cathy Warmund'
+ *                  avatarUrl: 'https://s3.amazonaws.com/uifaces/faces/twitter/geneseleznev/128.jpg'
+ *      401:
+ *        $ref: '#/components/responses/UnauthorizedError'
+ *      403:
+ *        $ref: '#/components/responses/UnauthorizedError'
```
Comment (wSedlacek) [inline api/order/orderRouter.js]: This needs to be updated for orders

File: api/order/orderRouter.js [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+ *      200:
+ *        description: A profile object
+ *        content:
+ *          application/json:
+ *            schema:
+ *              $ref: '#/components/schemas/Profile'
+ *      401:
+ *        $ref: '#/components/responses/UnauthorizedError'
+ *      404:
+ *        description: 'Profile not found'
```
Comment (wSedlacek) [inline api/order/orderRouter.js]: Same with this

File: api/order/orderRouter.js [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+ *          application/json:
+ *            schema:
+ *              type: object
+ *              properties:
+ *                message:
+ *                  type: string
+ *                  description: A message about the result
+ *                  example: profile created
+ *                profile:
+ *                  $ref: '#/components/schemas/Profile'
```
Comment (wSedlacek) [inline api/order/orderRouter.js]: This one too

File: api/order/orderRouter.js [hunk 4]
```diff
@@ -0,0 +1,297 @@
...
```
Comment (wSedlacek) [inline api/order/orderRouter.js]: And this one

File: api/order/orderRouter.js [hunk 5]
```diff
@@ -0,0 +1,297 @@
...
+      });
+    } catch (e) {
+      console.error(e);
+      res.status(500).json({ message: e.message });
+    }
+  } else {
+    res.status(404).json({ message: 'Order missing' });
+  }
+});
+/**
```
Comment (wSedlacek) [inline api/order/orderRouter.js]: We are missing a new line here

File: api/order/orderRouter.js [hunk 6]
```diff
@@ -0,0 +1,297 @@
...
+ *          application/json:
+ *            schema:
+ *              type: object
+ *              properties:
+ *                message:
+ *                  type: string
+ *                  description: A message about the result
+ *                  example: Profile '00uhjfrwdWAQvD8JV4x6' was deleted.
+ *                profile:
+ *                  $ref: '#/components/schemas/Profile'
```
Comment (wSedlacek) [inline api/order/orderRouter.js]: This one too

File: data/migrations/20201006184222_orders.js [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+  return knex.schema
+  .raw('CREATE EXTENSION IF NOT EXISTS "uuid-ossp"')
+  .createTable('orders', function(table) {
+      table.string('id').notNullable().unique().primary();
+      table.string('buyerId').notNullable();
+      table.integer('quantity').notNullable();
+      table.string('dateOrdered').notNullable();
+      table.string('status').notNullable();
+      table.integer('price');
+      table.boolean('priceDetermined').notNullable();//???
```
Comment (wSedlacek) [inline data/migrations/20201006184222_orders.js]: Yes this is not nullable, we can remove the comment.

File: api/profile/profileRouter.js [hunk 8]
```diff
@@ -167,7 +167,7 @@ router.post('/', authRequired, async (req, res) => {
       await Profiles.findById(id).then(async (pf) => {
         if (pf == undefined) {
           //profile not found so lets insert it
-          await Profiles.create(profile).then((profile) =>
+          await Profiles.create(profile).then((profile) =>//naming convention duplicated
```
Comment (wSedlacek) [inline api/profile/profileRouter.js]: We should remove this comment as there is no reason to change this file in this PR

**Review by wSedlacek — APPROVED** [review]

File: api/profile/profileRouter.js (hunk 8 above)
Comment (Shamskol) [reply api/profile/profileRouter.js]: Comment removed

File: api/order/orderRouter.js (hunk 1 above)
Comment (Shamskol) [reply api/order/orderRouter.js]: Updated

File: api/order/orderRouter.js (hunk 2 above)
Comment (Shamskol) [reply api/order/orderRouter.js]: Updated

File: api/order/orderRouter.js (hunk 3 above)
Comment (Shamskol) [reply api/order/orderRouter.js]: Updated

File: api/order/orderRouter.js (hunk 4 above)
Comment (Shamskol) [reply api/order/orderRouter.js]: Updated

File: api/order/orderRouter.js (hunk 6 above)
Comment (Shamskol) [reply api/order/orderRouter.js]: Updated

File: api/order/orderRouter.js [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+    })
+    .catch((err) => {
+      res.status(500).json({ error: err.message });
+    });
+});
+
+router.post('/', authRequired, async (req, res) => {
+  const newOrder = req.body;
+  if (newOrder) {
+    const id = newOrder.id || 0; //0 needs to be replaced with a string
```
Comment (wSedlacek) [inline api/order/orderRouter.js]: 0 needs to be replaced here before merging.
Can you look into `cuid` and the `uuid` libraries and use either one of those to replace this `0`?

File: api/order/orderRouter.js [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+    }
+  } else {
+    res.status(404).json({ message: 'Order missing' });
+  }
+});
+
+router.put('/', authRequired, function (req, res) {
+  const order = req.body;
+  if (order) {
+    const id = order.id || 0; // 0 needs to be replaced
```
Comment (wSedlacek) [inline api/order/orderRouter.js]: Here too

File: __tests__/routes/orders.test.js [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+//       };
+//       Profiles.findById.mockResolvedValue(profile);
+//       Profiles.update.mockResolvedValue([profile]);
+//       const res = await request(server).put('/profile/').send(profile);
+//       expect(res.status).toBe(200);
+//       expect(res.body.profile.name).toBe('Louie Smith');
+//       expect(Profiles.update.mock.calls.length).toBe(1);
+//     });
+//   });
+// });
```
Comment (wSedlacek) [inline __tests__/routes/orders.test.js]: Please either uncomment this code or delete it.
I will not merge Zombie code.

**Review by Istott — APPROVED** [review]
lgtm

**Pull request merged**